ladybug/designday.py

```diff
@@ -0,0 +1,1109 @@
+        for db in dry_bulb_condition.hourly_values:
+            if db >= max_dpt:
+                hourly_dew_point.append(max_dpt)
+            else:
+                hourly_dew_point.append(db)
+        return hourly_dew_point
+
+    def dew_point(self, db):
+        """Get the dew point (C), which is constant throughout the day (except at saturation).
+
+        args:
+            db: The maximum dry bulb temperature over the day.
+        """
+        if self._hum_type == 'Dewpoint':
+            return self._hum_value
+        elif self._hum_type == 'Wetbulb':
+            return dew_point_from_db_wb(
+                db, self._hum_value, self._barometric_pressure)
+        elif self._hum_type == 'HumidityRatio':
+            return dew_point_from_db_hr(
+                db, self._hum_value, self._barometric_pressure)
+        elif self._hum_type == 'Enthalpy':
+            return dew_point_from_db_enth(
+                db, self._hum_value / 1000,  self._barometric_pressure)
+
+    @property
+    def hourly_pressure(self):
+        """A list of barometric pressures for each hour over the design day."""
+        return [self._barometric_pressure for x in temp_multipliers]
+
+    @property
+    def hum_type(self):
+        """Get or set the humidity condition type."""
+        return self._hum_type
+
+    @hum_type.setter
+    def hum_type(self, data):
+        assert data in h_types, 'hum_type {} is not' \
+            ' recognized'.format(str(data))
+        self._hum_type = data
+
+    @property
+    def hum_value(self):
+        """Get or set the value of the humidity."""
+        return self._hum_value
+
+    @hum_value.setter
+    def hum_value(self, data):
+        assert isinstance(data, (float, int)), 'hum_value must be a' \
+            ' number. Got {}'.format(type(data))
+        self._hum_value = data
+
+    @property
+    def barometric_pressure(self):
+        """Get or set the value of the barometric pressure."""
+        return self._barometric_pressure
+
+    @barometric_pressure.setter
+    def barometric_pressure(self, data):
+        assert isinstance(data, (float, int)), 'barometric_pressure must be a' \
+            ' number. Got {}'.format(type(data))
+        self._barometric_pressure = data
+
+    @property
+    def isHumidityCondition(self):
+        """Return True."""
+        return True
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __repr__(self):
+        """humidity condition representation."""
+        return "HumidityCondition [{}: {}]".format(
+            self._hum_type, str(self._hum_value))
+
+
+class WindCondition(object):
+    """Represents wind and rain conditions on the design day.
+
+    attributes:
+        wind_speed
+        wind_direction
+        rain
+        snow_on_ground
+    """
+    def __init__(self, wind_speed, wind_direction=0,
+                 rain='No', snow_on_ground='No'):
```

Comment:
I would use True and False as default values and convert them to 'Yes', 'No' under the hood.

Reply:
Ok. I added a setter that automatically converts the boolean values to "Yes" and "No" under the hood.